zhmcclient/_resource.py

```diff
@@ -174,3 +174,29 @@ def get_property(self, name):
                 raise
             self.pull_full_properties()
             return self._properties[name]
+
+    def __str__(self):
+        """
+        Convert a BaseResource object to a string.
+        Used for printing.
+
+        example:
+        <Cpc _full_properties=False, _image_activation_profiles=None,
+        _load_activation_profiles=None, _lpars=None,
+        _manager=<zhmcclient._cpc.CpcManager object at 0x7f439cab3510>,
+        _partitions=None, _properties={u'status': u'service-required',
+        u'object-uri': u'/api/cpcs/f1bc49af-f71a-3467-8def-3c186b5d9352',
+        u'name': u'P0000P01'}, _properties_timestamp=1472558081,
+        _reset_activation_profiles=None,
+        _uri=/api/cpcs/f1bc49af-f71a-3467-8def-3c186b5d9352>
```

Comment:
This is not the usual syntax used for `__str__()` or  `__repr__()`.

The usual syntax used for `__repr__()` is a Python constructor syntax, e.g.:

    Cpc(_full_properties=False, _image_activation_profiles=None, ...)

Because `__str__()` is for human consumption, I don't think it makes a lot of sense to show all 100+ properties there. Maybe just name, status and object-uri?